--- a/mecab/config.sub
+++ b/mecab/config.sub
@@ -303,7 +303,8 @@
 	| v850 | v850e \
 	| we32k \
 	| x86 | xc16x | xstormy16 | xtensa \
-	| z8k | z80)
+  | z8k | z80 \
+  | aarch64)  # Chaquopy
 		basic_machine=$basic_machine-unknown
 		;;
 	c54x)
@@ -345,10 +346,11 @@
 	  basic_machine=$basic_machine-pc
 	  ;;
 	# Object if more than one company name word.
-	*-*-*)
-		echo Invalid configuration \`$1\': machine \`$basic_machine\' not recognized 1>&2
-		exit 1
-		;;
+  # Chaquopy has mor ethan one company word sometimes.
+	# *-*-*)
+	# 	echo Invalid configuration \`$1\': machine \`$basic_machine\' not recognized 1>&2
+	# 	exit 1
+	# 	;;
 	# Recognize the basic CPU types with company name.
 	580-* \
 	| a29k-* \
@@ -416,7 +418,8 @@
 	| x86-* | x86_64-* | xc16x-* | xps100-* \
 	| xstormy16-* | xtensa*-* \
 	| ymp-* \
-	| z8k-* | z80-*)
+  | z8k-* | z80-* \
+  | aarch64-*)  # Chaquopy
 		;;
 	# Recognize the basic CPU types without company name, with glob match.
 	xtensa*)
@@ -1483,6 +1486,8 @@
 		;;
 	-none)
 		;;
+  -clang) # Chaquopy
+    ;;
 	*)
 		# Get rid of the `-' at the beginning of $os.
 		os=`echo $os | sed 's/[^-]*-//'`
